reconcile/test/test_utils_gql.py

```diff
@@ -0,0 +1,72 @@
+import requests
+
+from reconcile.utils.gql import (
+    GqlApi,
+    GqlApiError,
+    GqlApiErrorForbiddenSchema,
+    GqlApiIntegrationNotFound,
+)
+from gql import Client
+from gql.transport.exceptions import TransportQueryError
+import pytest
+
+TEST_QUERY = """
+{
+    integrations: integrations_v1 {
+        name
+        description
+        schemas
+    }
+}
+"""
+
+
+def test_gqlapi_throws_gqlapierror_exception(mocker):
```

Comment:
I'd separate this out into separate tests rather than changing the mock along the way and making it a long test. Alternatively, you could look at pytest parameterize: https://docs.pytest.org/en/6.2.x/parametrize.html